google/cloud/aiplatform/models.py

```diff
@@ -215,10 +244,548 @@ def upload(
+                The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            description (str):
+                Optional. The description of the Endpoint.
+            traffic_split (Dict):
+                Optional. A map from a DeployedModel's ID to the
+                percentage of this Endpoint's traffic that
+                should be forwarded to that DeployedModel.
+                If a DeployedModel's ID is not listed in this
+                map, then it receives no traffic.
+
+                The traffic percentage values must add up to
+                100, or map must be empty if the Endpoint is to
+                not accept any traffic at a moment.
+            labels (Dict):
+                Optional. The labels with user-defined metadata to
+                organize your Endpoints.
+                Label keys and values can be no longer than 64
+                characters (Unicode codepoints), can only
+                contain lowercase letters, numeric characters,
+                underscores and dashes. International characters
+                are allowed.
+                See https://goo.gl/xmQnxf for more information
+                and examples of labels.
+            metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as
+                metadata.
+        Returns:
+            operation (lro.LRO):
+                Long-running operation of endpoint creation.
+        """
+        gapic_endpoint = gca_endpoint.Endpoint(
+            display_name=display_name,
+            description=description,
+            traffic_split=traffic_split,
+            labels=labels,
+        )
+
+        operation_future = api_client.create_endpoint(
+            parent=parent, endpoint=gapic_endpoint, metadata=metadata
+        )
+
+        return lro.LRO(operation_future)
+
+    def _allocate_traffic(
+        traffic_split: Sequence[gca_endpoint.Endpoint.TrafficSplitEntry],
+        traffic_percentage: int,
+    ) -> Dict:
+        """
+        Allocates desired traffic to new deployed model and scales traffic of
+        older deployed models.
+
+        Args:
+            traffic_split (Sequence[gca_endpoint.Endpoint.TrafficSplitEntry]):
+                Required. Current traffic split of deployed models in endpoint.
+            traffic_percentage (int):
+                Required. Desired traffic to new deployed model.
+        Returns:
+            traffic_split (Dict):
+                Traffic split to use.
+        """
+        old_models_traffic = 100 - traffic_percentage
+        if old_models_traffic:
+            unallocated_traffic = old_models_traffic
+            for deployed_model in traffic_split:
+                current_traffic = traffic_split[deployed_model]
+                new_traffic = int(current_traffic / 100 * old_models_traffic)
```

Comment:
does this guarantee that the sum of all percentages equals 100? it is required.

Reply:
the addition after the loop should guarantee it

Reply:
I see - please be sure to include a test for this.